### ISSUE TITLE
Execute configure script before running make on p2p

### ISSUE THREAD

**Issue by crioto** [subutai-io/snap#228]
Before building p2p we need to configure it. For this reason `configure` shell script needs to be executed, which generated `config.make` file used by Makefile.
`configure` accepting two parameters: 
* `--branch=<branch_name>`. This parameter can be omit - in this case configure script will try to determine branch manually by executing `git rev-parse --abbrev-ref HEAD`. Also note, that for release (production) build we need to pass `HEAD` as a value for this parameter. 
* `--dht=<dht_endpoint>`. This will set default DHT endpoint. Possible values are: `mdht.subut.ai:6881` for production, `54.93.172.70:6881` for master and `18.195.169.215:6881` for anything else including dev branch.

**Comment by lbthomsen** [comment]
Well yes :)  I am not sure what the issue is though?

**Comment by crioto** [comment]
It's an enhancement for snapcraft.

**Comment by lbthomsen** [comment]
Well - if you switch the plugin to "autotools" it will run ./configure ; make ; make install so that ought to work.  You can also specify customized ./configure parameters (see btrfs in the snapcraft.yaml.template)

**Comment by crioto** [comment]
So basically to make this work I need to modify this snapcraft.yaml.template? And what about docker image for tests? It is possible to use it?

**Comment by lbthomsen** [comment]
Yeah I just tested that - it should work as described at: https://github.com/lbthomsen/docker-snapcraft

If you're using Ubuntu just install snapcraft - that's better (certainly faster) but on an unsupported system the docker-snapcraft should work.

Simply - in your "snap" checkout run ```./configure``` and then ```snapcraft``` (using script that call docker).
